src/solvers/refinement/string_constraint_generator_transformation.cpp

```diff
@@ -359,38 +360,57 @@ string_exprt string_constraint_generatort::add_axioms_for_char_set(
   return res;
 }
 
+static optionalt<std::pair<exprt, exprt>> try_char_char(
```

Comment:
Not sure about use of pair. Consider using a struct with descriptive names for its data members.

Reply:
This function is local, static, documented and used only once. Adding `struct` where tuple just adds unnecessary boilerplate.